python/ray/data/datasource/file_based_datasource.py

```diff
@@ -60,6 +60,15 @@
 # 16 file size fetches from S3 takes ~1.5 seconds with Arrow's S3FileSystem.
 PATHS_PER_FILE_SIZE_FETCH_TASK = 16
 
+# The errors to retry for opening file.
+OPEN_FILE_RETRY_ON_ERRORS = ["AWS Error SLOW_DOWN"]
+
+# The max retry backoff in seconds for opening file.
+OPEN_FILE_RETRY_MAX_BACKOFF_SECONDS = 32
+
+# The max number of retry attempts for opening file.
+OPEN_FILE_RETRY_MAX_ATTEMPTS = 10
```

Comment:
Can we make these configurable by the user?

Reply:
I feel these configurations are advanced, so hesitate to put them in DataContext now. Users can change these Python variable directly, such as `FILE_SIZE_FETCH_PARALLELIZATION_THRESHOLD` and `PATHS_PER_FILE_SIZE_FETCH_TASK` above. I think we can add them into DataContext, if we see more users ask for this.